client/daemon.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+const daemonBinary = "dockerd"
+
+// DaemonProxy acts as a cli.Handler to proxy calls to the daemon binary
+type DaemonProxy struct{}
+
+// NewDaemonProxy returns a new handler
+func NewDaemonProxy() DaemonProxy {
+	return DaemonProxy{}
+}
+
+// CmdDaemon execs dockerd with the same flags
+// TODO: add a deprecation warning?
+func (p DaemonProxy) CmdDaemon(args ...string) error {
+	args = stripDaemonArg(os.Args[1:])
+
+	binaryAbsPath, err := exec.LookPath(daemonBinary)
+	if err != nil {
+		return err
+	}
+
+	return syscall.Exec(
```

Comment:
i'm afraid this won't work on Windows :( syscall.Exec is not implemented on windows. We should have tests covering `docker daemon`